integration_test/third_party_apps_test.go

```diff
@@ -327,6 +328,97 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
+// Returns the authoritative set of all apps available for testing.
+// The authoritative list corresponds to the directory names under
+// integration_test/third_party_apps_data/applications
+func determineAllApps(t *testing.T) map[string]bool {
+	allApps := make(map[string]bool)
+
+	files, err := os.ReadDir("third_party_apps_data/applications")
+	if err != nil {
+		t.Fatalf("got error listing files under third_party_apps_data/applications: %v", err)
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			allApps[file.Name()] = true
+		}
+	}
+	return allApps
+}
+
+func modifiedFiles(t *testing.T) []string {
+	cmd := exec.Command("git", "diff", "--name-only", "origin/master")
+	out, err := cmd.Output()
+	if err != nil {
+		t.Fatalf("got error calling `git diff`: %v", err)
+	}
+
+	return strings.Split(string(out), "\n")
+}
+
+// Determine what apps are impacted by current code changes.
+// Extracts app names as follows:
+//   apps/<appname>.go
+//   integration_test/third_party_apps_data/<appname>/
+// Checks the extracted app names against the set of all known apps.
+func determineImpactedApps(mf []string, allApps map[string]bool) map[string]bool {
+	impactedApps := make(map[string]bool)
+	for _, f := range mf {
+		if strings.HasPrefix(f, "apps/") {
+
+			// File names: apps/<appname>.go
+			f := strings.TrimPrefix(f, "apps/")
+			f = strings.TrimSuffix(f, ".go")
+
+			if _, ok := allApps[f]; ok {
+				impactedApps[f] = true
+			}
+		} else if strings.HasPrefix(f, "integration_test/third_party_apps_data/applications/") {
+			// Folder names: integration_test/third_party_apps_data/applications/<app_name>
+			f := strings.TrimPrefix(f, "integration_test/third_party_apps_data/applications/")
+			f = strings.Split(f, "/")[0]
+			// The directories here are already authoritative, no
+			// need to check against list.
+			impactedApps[f] = true
+
+		}
+	}
+	return impactedApps
+}
+
+type test struct {
+	platform   string
+	app        string
+	skipReason string
+}
+
+var defaultPlatforms = map[string]bool{
+	"debian-10":    true,
+	"windows-2019": true,
+}
+
+// Mark some tests for skipping, based on test_config and impacted apps.
+// Always test all apps against the default platform.  If a subset of apps is
+// determined to be impacted, also test all platforms for those apps.
+func determineTestsToSkip(tests []test, impactedApps map[string]bool, testConfig testConfig) {
+	for i, test := range tests {
+		if testing.Short() {
+			_, testApp := impactedApps[test.app]
+			_, defaultPlatform := defaultPlatforms[test.platform]
+			if defaultPlatform == false && testApp == false {
```

Comment:
Why not:

if !defaultPlatform && !testApp {

I think we usually prefer ! to == false, similar to preferring <nothing> to == true.